pkg/auth/docker_test.go

```diff
@@ -0,0 +1,84 @@
+package auth
+
+import (
+	"testing"
+
+	"github.com/docker/docker/api/types"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRegistryCredentialsLookupByHost(t *testing.T) {
+	correct := types.AuthConfig{Username: "correct"}
+	incorrect := types.AuthConfig{Username: "incorrect"}
+	tests := []struct {
+		name     string
+		regCreds RegistryCredentials
+		host     string
+	}{
+		{
+			name: "ExactMatch",
+			regCreds: RegistryCredentials{
+				"gcr.io":    correct,
+				"us.gcr.io": incorrect,
+			},
+			host: "gcr.io",
+		},
+		// TODO: In Christopher's code from prepushing.
```

Comment:
I don't think this is actually a thing outside of docker.io. I think we are safe to remove it.

Reply:
So do we need it for docker.io?

Reply:
Nevermind, I see how you handle it now.

Reply:
Yeah, that's basically what's tested on L39.

Reply:
delete this?